wasi-nn.witx

```diff
@@ -58,6 +63,63 @@
     (param $target $execution_target)
     (result $error (expected $graph (error $nn_errno)))
   )
+  ;;; Load an opaque sequence of bytes to use for inference.
+  ;;;
+  ;;; This allows runtime implementations to support multiple graph encoding formats. For unsupported graph encodings,
+  ;;; return `errno::inval`.
+  (@interface func (export "load_by_name")
+     ;;; The name of the model to load from the model registry
+     (param $model_name $utf8name)
+
+     (result $error (expected $graph (error $nn_errno)))
+  )
+
+  ;;; Allows registering a model via URI
+  ;;;
+  (@interface func (export "register_model_uri")
```

Comment:
My opinion would be to remove this method: it exposes the "load a model from a URI" idea to the Wasm context but I think that should only be available to the host. This kind of functionality would be useful in a host API (e.g., tell Wasmtime to load a model over HTTP to such and such a name) but it seems to open up the door for Wasm guests to do annoying or even malicious things, which seems bad.

Reply:
I agree. It's actually a reason for keeping register_named_model. Removed.